includes/class-reader-activation.php

```diff
@@ -959,21 +966,26 @@ public static function is_reader_verified( $user ) {
 	 * Warning: this method will only verify if the user is a reader in order to
 	 * authenticate. It will not check for any credentials.
 	 *
-	 * @param int $user_id User ID.
+	 * @param \WP_User|int $user_or_user_id User object.
 	 *
 	 * @return \WP_User|\WP_Error The authenticated reader or WP_Error if authentication failed.
 	 */
-	public static function set_current_reader( $user_id ) {
-		$user_id = \absint( $user_id );
-		if ( empty( $user_id ) ) {
-			return new \WP_Error( 'newspack_authenticate_invalid_user_id', __( 'Invalid user id.', 'newspack' ) );
+	public static function set_current_reader( $user_or_user_id ) {
+		if ( $user_or_user_id instanceof \WP_User ) {
+			$user = $user_or_user_id;
+		} elseif ( absint( $user_or_user_id ) ) {
+			$user = get_user_by( 'id', $user_or_user_id );
 		}
 
-		$user = \get_user_by( 'id', $user_id );
 		if ( ! $user || \is_wp_error( $user ) || ! self::is_user_reader( $user ) ) {
 			return new \WP_Error( 'newspack_authenticate_invalid_user', __( 'Invalid user.', 'newspack' ) );
 		}
 
+		$user_id = \absint( $user->ID );
+		if ( empty( $user_id ) ) {
+			return new \WP_Error( 'newspack_authenticate_invalid_user_id', __( 'Invalid user id.', 'newspack' ) );
+		}
+
```

Comment:
This check is superfluous, the user has already been validated on the previously.

Reply:
True, removed in 9d7f3ab.